pkg/kubectl/generate/versioned/deployment.go

```diff
@@ -64,7 +64,7 @@ func (b BaseDeploymentGenerator) structuredGenerate() (
 	}
 	podSpec = buildPodSpec(b.Images)
 	labels = map[string]string{}
-	labels["app"] = b.Name
+	labels["app.kubernetes.io/name"] = b.Name
```

Comment:
even though the recommended label is `app.kubernetes.io/name`
this change might break existing setups.